internal/provider/provider.go

```diff
@@ -2248,24 +2240,71 @@ func New(_ context.Context) (*schema.Provider, error) {
 		return configure(ctx, provider, d)
 	}
 
-	providerData := &conns.AWSClient{
-		// TODO: This should be generated.
-
-		// ServicePackageData is used before configuration to determine the provider's exported resources and data sources.
-		ServicePackages: []intf.ServicePackageData{
-			ec2.ServicePackageData,
-			globalaccelerator.ServicePackageData,
-			medialive.ServicePackageData,
-			meta.ServicePackageData,
-			resourceexplorer2.ServicePackageData,
-			simpledb.ServicePackageData,
-			sts.ServicePackageData,
-		},
+	var errs *multierror.Error
+	servicePackages := servicePackages(ctx)
+
+	for _, sp := range servicePackages {
+		for _, v := range sp.SDKDataSources(ctx) {
+			typeName := v.TypeName
+
+			if _, ok := provider.DataSourcesMap[typeName]; ok {
+				errs = multierror.Append(errs, fmt.Errorf("duplicate data source: %s", typeName))
+				continue
+			}
+
+			ds := v.Factory()
+
+			if v := ds.ReadWithoutTimeout; v != nil {
+				ds.ReadWithoutTimeout = wrappedReadContextFunc(v)
+			}
+
+			provider.DataSourcesMap[typeName] = ds
+		}
+	}
+
+	for _, sp := range servicePackages {
```

Comment:
Nit - Could this be combined with the range above (unless we intend to add multierror checking between)?